sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -304,4 +306,17 @@ static TableReference createTempTableReference(String projectId, String jobUuid)
         .setTableId(queryTempTableId);
     return queryTempTableRef;
   }
+
+  static String resolveTempLocation(
+      String tempLocationDir, String bigQueryOperationName, String stepUuid) {
+    try {
+      IOChannelFactory factory = IOChannelUtils.getFactory(tempLocationDir);
+      return factory.resolve(
+          factory.resolve(tempLocationDir, bigQueryOperationName),  stepUuid);
+    } catch (IOException e) {
+      throw new RuntimeException(
+          String.format("Failed to resolve temp destination directory in %s",
```

Comment:
This error message should include all the components, or at least the first two, to be actionable.